keps/sig-scheduling/2779-support-custom-queue-for-the-scheduler/README.md

```diff
@@ -0,0 +1,423 @@
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [x] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Provide an interface so that user can implement a custom queue of scheduler,
+with such a queue, user can sort the pods in the queue with specified rules 
+and get some functions that are not supported now.
+
+## Motivation
+The current queue (PriorityQueue in pkg\scheduler\internal\queue\scheduling_queue.go) 
+for pod selection in the scheduler is good enough for most cases, while it cannot handle 
+all the requests even with a custom scheduler based on the kubernetes scheduler plugins.
+
+One example:
+There are many pods from 2 users (userA and userB), and the target is to ensure resource
+usage (E.g. userA gets x CPUs) ratio between userA and userB is 1:1. Sorting the pods by
+the resource usage in Less function cannot work, because the resource usage for 
+userA/userB will be updated dynamically once a pod is selected and bound, while the 
+algorithm (heapsort) that used by the current queue to get the next pods depends on static
+data.
+
+Another example:
+The interval of function flushUnschedulableQLeftover is hard coded to 30 seconds, user
+cannot change it.
```

Comment:
It's now customizable, so no longer a valid motivation.

Reply:
So, can you compile at least one more motivation here?

Reply:
Update the message to:

Another requirement in terms of custom function support:  
For example, the interval of function flushUnschedulableQLeftover cannot be updated dynamically, while
user want it to be shorter or longer at different times as shorter interval can get pods to be scheduled
faster and longer one can get less logs flushed.

Reply:
Yeah, that's a fair point.